### PR TITLE
[COST-4111] Use source_type instead of provider_type in settings tag api

### PR DESCRIPTION
**Pull request by myersCody**
## Jira Ticket

[COST-4111](https://issues.redhat.com/browse/COST-41111)

## Description

This change will use source_type instead of provider type in the tag settings api. The `to_field_name` was not getting picked up, so I made some changes to check for it. 

## Testing

1. Checkout Branch
2. Restart Koku
3. Hit endpoint or launch shell
    1. You should see ...
4. Do more things...

## Notes

...

### CODE REVIEW

**Review by dlabrecq — DISMISSED** [review]

File: koku/api/settings/tags/serializers.py [hunk 1]
```diff
@@ -20,6 +20,12 @@ class SettingsTagSerializer(serializers.Serializer):
     class Meta:
         model = EnabledTagKeys
 
+    def to_representation(self, instance):
+        representation = super().to_representation(instance)
+        add_source_type = {"source_type": instance.provider_type, **representation}
+        add_source_type.pop("provider_type")
+        return add_source_type
```
Comment (maskarb) [inline koku/api/settings/tags/serializers.py:27]: This is a nitpick, but `to_representation` acts on the entire instance, not just the source-type. I would use something more generic than the `add_source_type` variable name, such as `data`:
```suggestion
    def to_representation(self, instance):
        data = super().to_representation(instance)
        data["source_type"] = data.pop("provider_type")
        return data
```

**Review by maskarb — APPROVED** [review]
Lgtm! Just one nitpick:

**Review by maskarb — APPROVED** [review]
lgtm!

**Pull request merged**